From c117f8edc471d3362043d88959d73c6a37e7e1e9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludovic=20Court=C3=A8s?= <ludo@gnu.org>
Date: Wed, 17 Jul 2024 00:00:00 +0200
Subject: build: Fix in-tree cross-compilation build.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Commit 57a889b7282dab303c4cdc49cccbbe22f961bd1c fixed out-of-tree
cross-compilation builds but broke in-tree cross-compilation builds.
With this change, we should have both.

* libguile/Makefile.am (gen-scmconfig.$(OBJEXT)): Use ‘-iquote’ instead
of ‘-I’.
---
 libguile/Makefile.am | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/libguile/Makefile.am b/libguile/Makefile.am
index 8d8fa27..ce97aba 100644
--- a/libguile/Makefile.am
+++ b/libguile/Makefile.am
@@ -62,15 +62,14 @@ gen_scmconfig_SOURCES = gen-scmconfig.c
 ## Override default rule; this should be compiled for BUILD host.  Note
 ## that we don't add $(AM_CPPFLAGS) here, as we need to run this
 ## program, but $(top_srcdir)/lib has a gnulib configured for the
-## target.  Instead we manually add $(top_builddir) and the current
-## directory, in order to pick up the generated config.h and
-## gen-scmconfig.h.  Nothing else from Guile is included by this code
-## generator.
+## target.  Instead we manually add $(top_builddir) and $(builddir), in
+## order to pick up the generated config.h and gen-scmconfig.h.  Nothing
+## else from Guile is included by this code generator.
 gen-scmconfig.$(OBJEXT): gen-scmconfig.c
 	$(AM_V_GEN) \
 	if [ "$(cross_compiling)" = "yes" ]; then \
 		$(CC_FOR_BUILD) $(DEFS) $(DEFAULT_INCLUDES) -I$(top_builddir) \
-		   -I. -c -o $@ $<; \
+		   -iquote$(builddir) -c -o $@ $<; \
 	else \
 		$(COMPILE) -c -o $@ $<; \
 	fi
-- 
cgit v1.1

